packages/preview-middleware-client/src/utils/version.ts

```diff
@@ -0,0 +1,15 @@
+import type { SingleVersionInfo } from '../../types/global';
+import VersionInfo from 'sap/ui/VersionInfo';
+import Log from 'sap/base/Log';
+
+/**
+ * Retrieve the UI5 version from sap.ui.core library
+ */
+export async function getUi5Version() {
```

Comment:
Could we return parsed version already? In many cases we need to know the minor version and then each time we parse it. Since we now have a common function, it would be convenient to have the parsing in the same place as well.

It could look something like this.
````typescript
interface UI5Version {
   full: string;
   major: number;
   minor: number;
   patch: number;
}

Reply:
I totally agree but I would not start the actual refactoring in this PR. This PR just wants to fix the undefined UI5 version handling bug.
I saw a parsing function in [adp/ui5-version-utils.ts](https://github.com/SAP/open-ux-tools/blob/main/packages/preview-middleware-client/src/adp/ui5-version-utils.ts) and there are quite some other functions that can be relocated to general utils as well. I just think it's not the scope of this PR to do it.

Reply:
ok, lets not get side tracked here. It seems that more changes would be required there to do this properly, lets have a separate PR for that.